include/ipr/impl

```diff
@@ -2092,7 +2092,7 @@ namespace ipr {
       private:
          void record_builtin_type(const ipr::As_type&);
 
-         using Filemap = stable_farm<impl::String>;
+         using Filemap = std::vector<const ipr::String*>;
```

Comment:
Is there a design where the string handles are stored directly?  Here, we basically have a vector of handles to handles...

Reply:
I considered storing impl::String, but it has two drawbacks:
1) we lose string unification
2) we would need to duplicates characters from ipr::String given as an argument to `File_index make_fileindex(const ipr::String&);`

Background: Currently IPR has two layered storage for strings. Characters are stored in the arena allocator. Then we have an impl::String layer in a red black tree for unification purposes that refers to util::string stored in the arena.

If we want strings that are reported from `const ipr::String& to_filename(File_index) const;` to be unified, we have to store the a reference/pointer to an impl::String stored in the red black tree.

Reply:
1.  Why is string unification required or needed for filename?

2.  Hmm, I am not sure I understand the need to duplicating the characters?

Reply:
The patch was based off inference of what was the intention behind:

make_fileindex, to_string(File_index) functions declared but not implemented in the lexicon.

Based on our discussion on 2020/02/05, I am removing those functions altogether and leaving management of filenames to the user of the IPR.